supervisor/shared/web_workflow/web_workflow.c

```diff
@@ -1179,7 +1209,9 @@ static void _process_request(socketpool_socket_obj_t *socket, _request *request)
                         request->authenticated = strncmp(request->header_value, prefix, strlen(prefix)) == 0 &&
                             strcmp(_api_password, request->header_value + strlen(prefix)) == 0;
                     } else if (strcasecmp(request->header_key, "Host") == 0) {
-                        request->redirect = strcmp(request->header_value, "circuitpython.local") == 0;
+                        // Do a prefix check so that port is ignored.
+                        const char *cp_local = "circuitpython.local";
```

Comment:
does it matter if a location like "circuitpython.local.example.com" would now be a match according to strncmp()?